test/drenv/commands.py

```diff
@@ -90,7 +90,8 @@ def run(*args, input=None, decode=True, env=None):
         try:
             p = subprocess.Popen(
                 args,
-                stdin=subprocess.PIPE if input else None,
+                # Avoid blocking foerver if there is no input.
+                stdin=subprocess.PIPE if input else subprocess.DEVNULL,
```

Comment:
I found that this does not work also on Linux, sudo does not care about stdin and will always try to get a password if not using -n. This should work for other programs that assume stdin, so it should be fine but I need more testing. I'll remove this commit from this PR for now.

Reply:
I verified that this works for programs that try to read from stdin. However there is no way to add a test for this, since pytest disable stdin in the same way, and there is no way to disable the feature.
https://github.com/pytest-dev/pytest/issues/2189